packages/abi-coder/src/Interface.ts

```diff
@@ -83,40 +83,34 @@ export class Interface<TAbi extends JsonAbi = JsonAbi> {
   }
 
   decodeLog(data: BytesLike, logId: number): any {
-    const { loggedType } = findOrThrow(this.jsonAbi.loggedTypes, (type) => type.logId === logId);
+    const loggedType = this.jsonAbi.loggedTypes.find((type) => type.logId === logId);
+    if (!loggedType) {
+      throw new FuelError(
+        ErrorCode.LOG_TYPE_NOT_FOUND,
+        `Log type with logId '${logId}' doesn't exist in the ABI.`
+      );
+    }
```

Comment:
This is something I mentioned in the last sync.

When dealing with multi-contracts ([see this](https://github.com/FuelLabs/fuels-ts/issues/1782)), users might call `ConractA`, which calls `ContractB`. Although `ContracA` will require an ABI, `ContractB` will not. So, if `ContractB` has some `log()` calls, those may bubble up to `ContractA`, which won't be able to decode them due to the lack of an ABI definition, and then we throw. The issue is that the transaction may have been successful, in which case throwing can be confusing.

Another way to handle this could be to print a warning instead of throwing an error when trying to decode `ContractB` logs as they come via `ContractA`. This approach could acknowledge the possible successful transaction while alerting the user to the decoding issue. It's worth considering but requires more in-depth thought to understand if it can cause other problems.

Feel free to leave this to another issue/PR.

---

@Torres-ssf Not sure how exactly this connects to the things you did in your PR. See if there's something else you'd want to add here.

Reply:
Do we know that the contract is external as the `id` of the [receipt](https://github.com/FuelLabs/fuels-ts/blob/master/packages/transactions/src/coders/receipt.ts) is non-zero.

We could disable throwing in these scenarios as this level, or a try/catch in the `getDecodedLogs` function.

Will leave this for a follow up issue.

Reply:
> It's worth considering but requires more in-depth thought to understand if it can cause other problems.

Added a follow up [issue](https://github.com/FuelLabs/fuels-ts/issues/2004) for this 😄